pkg/kubelet/pod/mirror_client.go

```diff
@@ -82,6 +85,16 @@ func (mc *basicMirrorClient) DeleteMirrorPod(podFullName string) error {
 		klog.Errorf("Failed to parse a pod full name %q", podFullName)
 		return err
 	}
+	if uid != nil {
+		if apiPod, err := mc.apiserverClient.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{}); err == nil {
+			klog.Infof("DeleteMirrorPod %s: requested UID %s actual %s", podFullName, *uid, apiPod.ObjectMeta.UID)
+			if apiPod.ObjectMeta.UID != *uid {
+				return fmt.Errorf("UID of pod %q (%s) does not match requested UID (%s); not deleting mirror pod", podFullName, apiPod.ObjectMeta.UID, *uid)
+			}
+		} else {
+			return fmt.Errorf("Unable to retrieve pod %s", podFullName)
```

Comment:
Should err be included in the message ?

Reply:
also we should probably say "mirror pod"

Reply:
Yes and yes.

Reply:
This goes away; it's all in DeleteMirrorPod now.

Reply:
Fixed.